client/xamarin.forms/ZUMOAPPNAME/TodoItemManager.cs

```diff
@@ -22,71 +24,81 @@ public TodoItemManager()
 				Constants.GatewayURL,
 				Constants.ApplicationKey);
 
-			this.todoTable = client.GetTable<TodoItem>();
+            var store = new MobileServiceSQLiteStore("localstore.db");
+            store.DefineTable<TodoItem>();
 
-		}
-//		public ToDoItem GetTaskFromList(string id)
-//		{
-//			return todoTable.FirstOrDefault(o => o.ID == id);   
-//		}
-		public async Task<TodoItem> GetTaskAsync(string id)
-		{
-			try
-			{
-				return await todoTable.LookupAsync(id);
-			}
-			catch (MobileServiceInvalidOperationException msioe)
-			{
-				Debug.WriteLine(@"INVALID {0}", msioe.Message);
-			}
-			catch (Exception e)
-			{
-				Debug.WriteLine(@"ERROR {0}", e.Message);
-			}
-			return null;
-		}
-		public async Task<ObservableCollection<TodoItem>> GetTodoItemsAsync()
-		{
-			try
-			{
-				return new ObservableCollection<TodoItem>(await todoTable.ReadAsync());
-			}
-			catch (MobileServiceInvalidOperationException msioe)
-			{
-				Debug.WriteLine(@"INVALID {0}", msioe.Message);
-			}
-			catch (Exception e)
-			{
-				Debug.WriteLine(@"ERROR {0}", e.Message);
-			}
-			return null;
-		}
-		public async Task SaveTaskAsync(TodoItem item)
-		{
-			if (item.ID == null)
-			{
-				await todoTable.InsertAsync(item);
-				//TodoViewModel.TodoItems.Add(item);
-			}
-			else
-				await todoTable.UpdateAsync(item);
-		}
-		public async Task DeleteTaskAsync(TodoItem item)
-		{
-			try
-			{
-				//TodoViewModel.TodoItems.Remove(item);
-				await todoTable.DeleteAsync(item);
-			}
-			catch (MobileServiceInvalidOperationException msioe)
-			{
-				Debug.WriteLine(@"INVALID {0}", msioe.Message);
-			}
-			catch (Exception e)
-			{
-				Debug.WriteLine(@"ERROR {0}", e.Message);
-			}
-		}
-	}
+            //Initializes the SyncContext using the default IMobileServiceSyncHandler.
+            this.client.SyncContext.InitializeAsync(store);
+
+            this.todoTable = client.GetSyncTable<TodoItem>();
+        }
+
+        public async Task SyncAsync()
+        {
+            await this.todoTable.PullAsync("todoItems", this.todoTable.CreateQuery());
```

Comment:
Added some basic error handling to bring it to parity with what we have for iOS.